### ISSUE TITLE
Pressing Space in an InputText causes a Enter press (and enter press makes page go back to previous page)

### ISSUE THREAD

**Issue by tobiazsh** [FlorianMichael/fabric-imgui-example-mod#11]
When pressing space or enter on the keyboard in ImGui, the page jumps back once although it shouldn't. This is especially annoying in ImGui.inputText's since I cannot press space without it going back.

Example code;
```java
ImGui.begin("Test");

ImGui.inputText(String, ImString);

ImGui.end();
```

This issue has already been addressed in the official ImGui repo and marked as fixed but I do not know how to implement their fix in java.

https://github.com/ocornut/imgui/issues/4552

I see that InputTextFlags.EnterReturnsTrue has been added but that does the exact opposite of what I want to see.

If anyone could help me on this one, that'd be great! Thanks!

**Comment by florianreuth** [comment]
You will probably have to wait for ImGui-Java to update their included imgui version to get the fix as well

**Comment by tobiazsh** [comment]
I mean, the version should already be the one with the fix, or at least it says that on the about menu. Funny enough, it works just fine in the about window.

**Comment by tobiazsh** [comment]
ImGui was not at fault. So, what basically happend is that the button in the background from the Minecraft GUI that opens the ImGui GUI was still focused and whenever you pressed Enter or Space, it got pressed and the ImGui Window opened again thus going back to the first window. I noticed this when I tried if the same thing happens in an actual Minecraft Instance instead of the development environment (where I had sounds turned off) and I heard the typical Minecraft button "click" sound. By then, I knew something was off and I checked the code. Turns out just removing all the children from the Minecraft Screen was not enough and I also had to defocus any focused element to prevent it from being clicked. I've done this by just re-drawing everything so it loses focus and then removing the children. screen.blur(); was not enough apparently.

TL;DR:
ImGui works fine. Minecraft button was still focused, whenever space or enter was pressed, it activated. By re-drawing everything, it loses focus and then works fine.

**Comment by florianreuth** [comment]
Oh I see, I was already confused but that makes sense, thanks for clarification and sorry for the long response times.
